### ISSUE TITLE
BiglyBT freeze with (phantom?) torrent

### ISSUE THREAD

**Issue by SUCM** [BiglySoftware/BiglyBT#2134]
for unknown reason, starting few hours ago, BiglyBT UI would freeze forever whenever I try to (scroll, down arrow key, page down ...etc) display the last torrent at the bottom of Library. Bigly doesn't freeze if I scroll to just show half of the last item, but as soon as I do whatever action to make last torrent show up in UI, Bigly freezes. this also happens if I drag/adjust panel size to display the last item

eventually, I played some add/del torrent thing to make the problematic torrent showed up, and surprisingly, it's displaying the correct alcohol 120% file type icon. it seems that Bigly trying to display alcohol 120% file type icon for mdf file on download completion would cause freeze and corrupting Bigly torrent list. I thought doing restore to a date without the problematic torrent would help, but it looks like the problematic torrent stays despite not counted even with restore. I tried to edit Bigly's downloads.config to remove the problematic torrent, but that also didn't work. how do I effectively edit or remove a torrent from library if Bigly always freeze and editing downloads.config doesn't remove the torrent?

since Bigly is able to display alcohol 120% file type icon properly with some clicks and torrent add/remove actions, it seems that the freeze when trying to display icons may not be alcohol 120%'s problem?

as part of the torrent add/remove. I see a weird behaviour. I started with Bigly counting 60 torrents, but the problematic torrent is the 61th torrent. the problematic torrent only showed up as the 60th torrent when I deleted 1 torrent and it showed up with the correct alcohol 120% file type icon. at this point, I can scroll up down left right without problem and alcohol 120% file type icon display working just fine in Bigly

**Comment by parg** [comment]
Disable the icon (right-cick the 'name' column header), delete it and then re-enable?

No idea about why the problem occurs although who knows what shit might be triggered by Windows when dealing with the icon...

**Comment by SUCM** [comment]
does Bigly cache icon/file? after one of the trial and error when trying to recover from the always freeze , I deleted the downloaded files from HDD. however, Bigly still freezes when relaunch with phantom torrent. the other thing is that Bigly doesn't even count the torrent until I deleted one of the existing normal torrent. as I was able to scroll to bottom and show only half of the last torrent, Bigly wasn't even going to display the problematic torrent. hence not sure about the freeze

forgot to mention that before I did the restore, I didn't need to scroll to bottom to hit freeze. just doing any scroll in the Library panel would freeze UI. everything worked when not doing scrolling and keep list at the very top. after restore, I need to scroll to the very bottom of the list to hit freeze. download and completed lists are both sorted in order, so oldest torrent are at the top. hence, this part didn't make sense to me

this is only one instance of torrent causing Bigly freeze. previously, there were 2 another torrents causing similar freeze on Bigly startup even before UI init completes (library panel not yet shown/drawn), I didn't even have any chance to scroll. after few days of debug, I feel they were probably due to abnormal long and deep (lots of long name sub-directories) paths. since those 2 torrents happened when deleting or modifying path in downloads.config was working, I simply deleted them in downloads.config to recover. however, Bigly continue to freeze right away when those torrents were added back. freeze happened when torrent gets into list, not when dealing with new torrent's window/panel. it took me few days to figure out for those 2 since I had never selected any file in the long and deep path, but Bigly still created the complete directory structure with empty directories
